### PR TITLE
Add simorgh

### PR DESCRIPTION
**Pull request by nilchia**
Add a new Galaxy tool wrapper for SIMORGH, a method for predicting apo-state and cryptic ligand binding sites in protein structures. 

FOR CONTRIBUTOR:
* [x] I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] License permits unrestricted use (educational + commercial)
* [x] This PR adds a new tool or tool collection
* [ ] This PR updates an existing tool or tool collection
* [ ] This PR does something else (explain below)

There are two labels that allow to ignore specific (false positive) tool linter errors:

* `skip-version-check`: Use it if only a subset of the tools has been updated in a suite.
* `skip-url-check`: Use it if github CI sees 403 errors, but the URLs work.

### CODE REVIEW

**Comment by nilchia** [comment]
@amisteromid; This is the first draft.

Please check if the parameters are explained correctly and the pipeline is correct.

**Comment by nilchia** [comment]
The container has not yet been pushed to Docker Hub. I'll update the requirement once it is available online.

File: tools/simorgh/test-data/multi_model.pdb
Comment (amisteromid) [inline tools/simorgh/test-data/multi_model.pdb:1]: This is a multi-model PDB.  I'll send you a single one as well to include

File: tools/simorgh/macros.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        <requirements>
+            <container type="docker">
+                simorgh_updated_v1:latest
+            </container>
+        </requirements>
+    </xml>
+    <xml name="creators">
+        <creator>
+            <person givenName="Amirhossein" familyName="Naghsh Nilchi" email="nilchia@informatik.uni-freiburg.de" identifier="https://orcid.org/0000-0003-1329-3044"/>
+            <person givenName="Omid" familyName="mokhtari" email="omid.mokhtari@inria.fr" identifier="https://orcid.org/0009-0000-8751-8117"/>
```
Comment (amisteromid) [inline tools/simorgh/macros.xml]:
```suggestion
            <person givenName="Omid" familyName="Mokhtari" email="omid.mokhtari@inria.fr" identifier="https://orcid.org/0009-0000-8751-8117"/>
            <person givenName="Hamed" familyName="Khakzad" email="hamed.khakzad@inria.fr" identifier="https://orcid.org/0000-0002-8556-0650"/>
            
```

File: tools/simorgh/simorgh.xml [hunk 2]
```diff
@@ -0,0 +1,219 @@
+<tool id="simorgh" name="SIMORGH" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description>Apo-State and Cryptic Ligand Binding Site Prediction</description>
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml]:
```suggestion
    <description>Deep learning framework for apo-state and cryptic pocket prediction</description>
```

File: tools/simorgh/simorgh.xml [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    <inputs>
+        <param name="pdb" type="data" format="pdb" label="PDB file"/>
+        <conditional name="bbflow_condi">
+            <param name="bbflow" type="select" label="Run BBflow?" help="It runs a structure-based conformational ensemble prediction with flow matching">
+                <option value="no">No</option>
+                <option value="yes">Yes</option>
+            </param>
+            <when value="no"/>
+            <when value="yes">
+                <param argument="--num_samples" type="integer" value="100" min="1" label="Number of conformations to sample"/>
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml]: default better be 25, 100 takes too long and seems unnecessary.

File: tools/simorgh/simorgh.xml [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+alternative backbone conformations with BBflow before scoring — this can
+improve sensitivity to pockets that are not visible in the input structure
+as given, at the cost of longer runtime.
+
+Predicted site scores can optionally be grouped into discrete pockets with
+HDBSCAN density-based clustering, which is useful when you want candidate
+binding sites as separate clusters rather than a per-residue/per-point
+probability map.
+
+Inputs
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml]:
```suggestion
SIMORGH predicts apo-state and cryptic ligand-binding sites from a single protein structure. Given an input PDB file, it outputs a residue-level predicted probability of ligand binding for each residue, using two trained models — a geometric encoder and an ensemble aggregator — and writes the predictions to an output PDB file.

Apo-state structures lack ligand-induced pocket organization, which poses a bigger challenge for pocket prediction. This challenge is more pronounced for cryptic pockets. To address this, we can optionally generate an ensemble of alternative backbone conformations using BBflow, capturing greater conformational heterogeneity and improving sensitivity to pockets not visible in the input structure.

Predicted site scores can also optionally be clustered using HDBSCAN — useful when you want to detect multiple binding sites separately and remove the noise, rather than just a general per-residue probability map.


Inputs
```

File: tools/simorgh/simorgh.xml [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+Outputs
+--------
+
+* **Predicted Structure**: the input structure (or the BBflow-sampled
+  conformation, if BBflow was run) annotated with predicted binding-site
+  scores.
+* **BBflow Structure**: only produced if BBflow sampling was enabled; the
+  backbone conformation(s) sampled by BBflow prior to prediction.
+    ]]></help>
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml:436]:
```suggestion
* **Predicted Structure**: the input structure (or the BBflow-sampled
  conformation, if BBflow was run) annotated with predicted binding-site
  scores in b-factor column.
* **BBflow Structure**: only produced if BBflow sampling was enabled; the
  backbone conformations sampled by BBflow prior to prediction.
    ]]></help>
```

File: tools/simorgh/.shed.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+name: simorgh
+owner: bgruening
+description: Apo-State and Cryptic Ligand Binding Site Prediction
+homepage_url: https://github.com/amisteromid/SIMORGH
+long_description: |
+  SIMORGH is a deep learning-based method for predicting apo-state and cryptic ligand binding sites in proteins. It leverages a combination of convolutional neural networks and graph-based representations to accurately identify potential binding pockets, even in the absence of bound ligands. This tool is particularly useful for drug discovery and protein engineering applications, where understanding the dynamic nature of protein structures is crucial.
```
Comment (amisteromid) [inline tools/simorgh/.shed.yml]:
```suggestion
  SIMORGH is a deep learning-based method for predicting apo-state and cryptic ligand binding sites in proteins. It leverages SE(3) equivariant neural networks to accurately identify potential binding pockets. This tool is particularly useful for drug discovery, where understanding the dynamic nature of protein structures is crucial.
```

File: tools/simorgh/.shed.yml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+remote_repository_url: https://github.com/bgruening/galaxytools/tree/master/tools/simorgh
+type: unrestricted
+categories:
+  - Computational chemistry
+  - Molecular Dynamics
+  - Structural Materials Analysis
+auto_tool_repositories:
```
Comment (amisteromid) [inline tools/simorgh/.shed.yml]:
```suggestion
  - Molecular Dynamics
auto_tool_repositories:
```

File: tools/simorgh/test-data/multi_model.pdb
Comment (nilchia) [reply tools/simorgh/test-data/multi_model.pdb:1]: single model PDB is included now.

File: tools/simorgh/simorgh.xml [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+    BBflow generates.
+  * **Number of sampling timesteps**: number of integration steps used by
+    the BBflow sampler; more timesteps can improve sample quality at the
+    cost of runtime.
+  * **Batch size**: number of conformations sampled per batch; left blank,
+    it is estimated automatically.
+  * **Advanced prior parameters (gamma_trans / gamma_rots)**: control how
+    closely the sampling prior sticks to the input (equilibrium) structure.
+    Left blank, the values from the model's training configuration are used.
+
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml:403]:
```suggestion
    Left blank, the values from the model's training configuration are used.
  * **Tag of the model to use**: checkpoint/architecture variant for BBflow 
      sampling (e.g., single-chain vs. multimer versions).

  * **Create visualizations?**: if enabled, generates an interactive 3D HTML 
    visualization showing predicted binding probabilities and clusters.

    * **Visualization Style**: molecular representation style used for the 
      structure rendering (sphere, surface, or cartoon).
```

File: tools/simorgh/simorgh.xml [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+    the distance metric used for clustering.
+
+Outputs
+--------
+
+* **Predicted Structure**: the input structure (or the BBflow-sampled
+  conformation, if BBflow was run) annotated with predicted binding-site
+  scores in b-factor column.
+* **BBflow Structure**: only produced if BBflow sampling was enabled; the
+  backbone conformations sampled by BBflow prior to prediction.
```
Comment (amisteromid) [inline tools/simorgh/simorgh.xml:429]:
```suggestion
* **BBflow Structure**: only produced if BBflow sampling was enabled; the
  backbone conformations sampled by BBflow prior to prediction.
* **Predicted Structure, clusters**: only produced if clustering is enabled; 
  a PDB structure containing the cluster assignments.
* **Visualization**: only produced if visualization is enabled; an interactive 
  standalone HTML file for inspecting residue predictions and clustered pockets.
```

**Review by amisteromid — APPROVED** [review]

File: tools/simorgh/.shed.yml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+name: simorgh
+owner: bgruening
+description: Apo-State and Cryptic Ligand Binding Site Prediction
+homepage_url: https://github.com/amisteromid/SIMORGH
```
Comment (pavanvidem) [inline tools/simorgh/.shed.yml:4]: Is this link still working?

File: tools/simorgh/macros.xml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    <xml name="creators">
+        <creator>
+            <person givenName="Amirhossein" familyName="Naghsh Nilchi" email="nilchia@informatik.uni-freiburg.de" identifier="https://orcid.org/0000-0003-1329-3044"/>
+            <person givenName="Omid" familyName="Mokhtari" email="omid.mokhtari@inria.fr" identifier="https://orcid.org/0009-0000-8751-8117"/>
+            <person givenName="Hamed" familyName="Khakzad" email="hamed.khakzad@inria.fr" identifier="https://orcid.org/0000-0002-8556-0650"/>
+            <person givenName="Björn" familyName="Grüning" email="gruening@informatik.uni-freiburg.de" identifier="https://orcid.org/0000-0002-3079-6586"/>
+        </creator>
+    </xml>
+    <xml name="edam">
+        <edam_topics>
```
Comment (pavanvidem) [inline tools/simorgh/macros.xml]: If you can acces the bio.tools entry, please add the edam terms to the bio.tools entry instead of here.

File: tools/simorgh/simorgh.xml [hunk 12]
```diff
@@ -0,0 +1,402 @@
...
+        export DEVICE=`python -c "import torch; print('cuda' if torch.cuda.is_available() else 'cpu')"`
+        && echo "Using device: \$DEVICE"
+        && ln -s '$pdb' input.pdb
+        #if str($bbflow_condi.bbflow) == 'yes':
+            && bbflow_sample
+                --input_path input.pdb
+                --output_path bbflow.pdb
+                --output_format pdb
+                --num_samples $bbflow_condi.num_samples
+                #if str($bbflow_condi.batch_size) != '':
```
Comment (pavanvidem) [inline tools/simorgh/simorgh.xml:23]: For a non optional param, it cannot be empty.

File: tools/simorgh/simorgh.xml [hunk 13]
```diff
@@ -0,0 +1,402 @@
...
+                    --batch_size $bbflow_condi.batch_size
+                #end if
+                --device \$DEVICE
+                --timesteps $bbflow_condi.timesteps
+                --tag $bbflow_condi.advanced.tag
```
Comment (pavanvidem) [inline tools/simorgh/simorgh.xml]:
```suggestion
                --tag '$bbflow_condi.advanced.tag'
```

File: tools/simorgh/simorgh.xml [hunk 14]
```diff
@@ -0,0 +1,402 @@
...
+                #if str($cluster_condi.cluster) == 'yes':
+                    --cluster
+                    --cutoff $cluster_condi.cutoff
+                    --min-cluster-size $cluster_condi.min_cluster_size
+                    --min-samples $cluster_condi.min_samples
+                    --alpha $cluster_condi.alpha
+                #end if
+                #if str($viz_condi.visualize) == 'yes':
+                    --visualize
+                    --viz-style "$viz_condi.viz_style"
```
Comment (pavanvidem) [inline tools/simorgh/simorgh.xml]:
```suggestion
                    --viz-style '$viz_condi.viz_style'
```

**Review by pavanvidem — COMMENTED** [review]
If there is no bio.tools entry, please create one.

File: tools/simorgh/simorgh.xml (hunk 12 above)
Comment (nilchia) [reply tools/simorgh/simorgh.xml:23]: But it is [optional](https://github.com/nilchia/galaxytools/blob/4cd79cc17b179461df49fbf4a718269c6b1e88e0/tools/simorgh/simorgh.xml#L68):
```
                <param argument="--batch_size" type="integer" optional="true" min="1" label="Batch size" help="Leave blank to estimate automatically"/>
```
no?

**Comment by nilchia** [comment]
Please do not merge, @amisteromid needs to double check the license with his institute.

File: tools/simorgh/macros.xml (hunk 11 above)
Comment (nilchia) [reply tools/simorgh/macros.xml]: I added in bio tools. Should I remove it from here, or should it be in both?

Comment (nilchia) [reply tools/simorgh/macros.xml]: Tool is added to bio.tools and the edams are removed.

File: tools/simorgh/.shed.yml (hunk 10 above)
Comment (amisteromid) [reply tools/simorgh/.shed.yml:4]: Hey! We're confirming the license with my institute. It'll be public soon...

**Comment by nilchia** [comment]
It is good to go from my side
@pavanvidem 
@bgruening 

The lint is failing because the repo is private (I put the link in help section). It will get public before submission as @amisteromid mentioned.

**Comment by bgruening** [comment]
I added skip-url-check to when you merge it will be deployed even with this link failure

**Review by pavanvidem — APPROVED** [review]

**Comment by pavanvidem** [comment]
Thanks @amisteromid @nilchia

**Comment by nilchia** [comment]
Thank you @pavanvidem  and @bgruening :pray:
